includes/Core/Authentication/Clients/Google_Proxy_Client.php

```diff
@@ -140,6 +141,37 @@ public function revokeToken( $token = null ) {
 		return 200 === (int) $response->getStatusCode();
 	}
 
+	/**
+	 * Adds auth listeners to the HTTP client based on the credentials set in the Google API Client object.
+	 *
+	 * @since 1.0.3
+	 *
+	 * @param ClientInterface $http The HTTP client object.
+	 * @return ClientInterface The HTTP client object
+	 */
+	public function authorize( ClientInterface $http = null ) {
```

Comment:
Why is this method overridden? It looks like the parent method would still refresh the token and call the `token_callback` when needed.

Reply:
The overriding version I created will refresh the token the proper way (using the proxy), so when the parent method is called, the token would no longer be "expired".

The reason I'm overriding this is that, whenever an actual Google API request would be attempted by the plugin with an expired access token, it would fail because that part of the Google client library had not yet been adjusted to use the proxy instead of the original OAuth endpoint.